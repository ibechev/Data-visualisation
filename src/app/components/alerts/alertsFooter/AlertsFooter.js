import React from "react";
import PropTypes from "prop-types";

import Button from "../../button/Button";

const AlertsFooter = ({ handleSeeMore, handleViewAll }) => {
  return (
    <footer className="alerts-footer">
      <Button bgc="primary-light" onClick={handleSeeMore}>
        See more <i className="fas fa-caret-down" />
      </Button>
      <Button noBody onClick={handleViewAll}>
        View all
      </Button>
    </footer>
  );
};

AlertsFooter.propTypes = {
  handleSeeMore: PropTypes.func.isRequired,
  handleViewAll: PropTypes.func.isRequired
};

export default AlertsFooter;
