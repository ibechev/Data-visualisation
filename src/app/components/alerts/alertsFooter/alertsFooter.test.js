import React from "react";
import { mount } from "enzyme";
import AlertsFooter from "./AlertsFooter";
import renderer from "react-test-renderer";
import { propTypeErrors } from "../../../../../tests/config/testUtils";

describe("Expect AlertsFooter", () => {
  const defaultProps = {
    handleSeeMore: () => {},
    handleViewAll: () => {}
  };

  const setup = (testProps = {}) => {
    const props = {
      ...defaultProps,
      ...testProps
    };
    return mount(<AlertsFooter {...props} />);
  };

  it("to render correctly", () => {
    const wrapper = renderer
      .create(<AlertsFooter {...defaultProps} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to not throw error with correct props", () => {
    propTypeErrors(<AlertsFooter {...defaultProps} />, defaultProps);
  });

  it("to call button handlers on each click of a button", () => {
    const spyBtnHandler = jest.fn();
    const testProps = {
      handleSeeMore: spyBtnHandler,
      handleViewAll: spyBtnHandler
    };
    const wrapper = setup(testProps);

    wrapper.find("button").forEach(button => button.simulate("click"));
    expect(spyBtnHandler).toHaveBeenCalledTimes(wrapper.find("button").length);
  });
});
