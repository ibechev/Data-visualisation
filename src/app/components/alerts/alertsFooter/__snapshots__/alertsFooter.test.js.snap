// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect AlertsFooter to render correctly 1`] = `
<footer
  className="alerts-footer"
>
  <button
    className="button primary-light"
    onClick={[Function]}
    type="button"
  >
    See more 
    <i
      className="fas fa-caret-down"
    />
  </button>
  <button
    className="button no-body"
    onClick={[Function]}
    type="button"
  >
    View all
  </button>
</footer>
`;
